server/src/main/java/org/opensearch/indices/replication/copy/SegmentReplicationPrimaryService.java

```diff
@@ -237,7 +237,13 @@ public void messageReceived(TrackShardRequest request, TransportChannel channel,
             final StepListener<ReplicationResponse> addRetentionLeaseStep = new StepListener<>();
             final Consumer<Exception> onFailure = e -> {
                 assert Transports.assertNotTransportThread(this + "[onFailure]");
-                logger.error("Failure", e);
+                logger.error("Error marking shard {} as tracked for allocation ID {}, {}", shardId, request.getTargetAllocationId(), e);
```

Comment:
Nit - This seems wonky. Did you mean to print the exception string rather than dump the stack trace?

Reply:
oops good catch. Updated with ParameterizedMessage.